### PR TITLE
FEAT: EDB Configuration support in SIwave

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

**Comment by ring630** [comment]
CI failure due to bug https://github.com/actions/upload-artifact/issues/591

**Comment by SMoraisAnsys** [comment]
Updating the branch after main fix

File: src/pyedb/siwave.py [hunk 1]
```diff
@@ -37,6 +41,29 @@ def wait_export_file(flag, file_path, time_sleep=0.5):
...
+            for filename in filenames:
+                file_path = os.path.join(dirpath, filename)
+                if os.path.isfile(file_path):
+                    total_size += os.path.getsize(file_path)
+
+        if total_size > 0:
+            break
+        else:
+            time.sleep(time_sleep)
+    return True
```
Comment (SMoraisAnsys) [inline src/pyedb/siwave.py:64]:
```suggestion
    while True:
        if os.path.exists(folder_path):
            for root, _, files in os.walk(folder_path):
                if any(os.path.getsize(os.path.join(root, file)) > 0 for file in files):
                    return True
        
        # Wait before checking again.
        time.sleep(time_sleep)
```

File: src/pyedb/siwave.py [hunk 2]
```diff
@@ -425,3 +452,89 @@ def run_icepak_simulation(self, icepak_simulation_name, dc_simulation_name):
...
+        bool
+        """
+        if isinstance(file_path, Path):
+            file_path = str(file_path)
+        flag = self.oproject.ScrExportEDB(file_path)
+        if flag == 0:
+            self._logger.info(f"Exporting EDB to {file_path}.")
+            return wait_export_folder(flag, file_path, time_sleep=1)
+        else:  # pragma no cover
+            raise f"Failed to export EDB to {file_path}."
```
Comment (SMoraisAnsys) [inline src/pyedb/siwave.py:475]:
```suggestion
            raise RuntimeError(f"Failed to export EDB to {file_path}.")
```

File: src/pyedb/siwave.py [hunk 3]
```diff
@@ -425,3 +452,89 @@ def run_icepak_simulation(self, icepak_simulation_name, dc_simulation_name):
...
+        bool
+        """
+        if isinstance(file_path, Path):
+            file_path = str(file_path)
+        flag = self.oproject.ScrImportEDB(file_path)
+        if flag == 0:
+            self._logger.info(f"Importing EDB to {file_path}.")
+            return True
+        else:  # pragma no cover
+            raise f"Failed to export EDB to {file_path}."
```
Comment (SMoraisAnsys) [inline src/pyedb/siwave.py:495]:
```suggestion
            raise RuntimeError(f"Failed to import EDB to {file_path}.")
```

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
I left minor changes (refactoring and fixing typo)

**Comment by ring630** [comment]
@SMoraisAnsys thanks for the comments

**Review by SMoraisAnsys — APPROVED** [review]
LGTM

**Pull request merged**